File: day-1/calorie_counting.rb
# frozen_string_literal: true

ELVES = File.read('input.txt').split("\n\n").freeze
TOTAL_CALORIES = ELVES.map { |elf_calories| elf_calories.split.map(&:to_i).sum }.freeze

def solve_first_puzzle
  TOTAL_CALORIES.max
end

def solve_second_puzzle
  TOTAL_CALORIES.max(3).sum
end

solve_first_puzzle
solve_second_puzzle

File: day-1/calorie_counting.py
input = open("input.txt").readlines()

elves = {}
current_elf = 0

for item in input:
  if item == '\n':
    current_elf += 1
    continue
  else:
    value = int(item.replace('\n', ''))
    if current_elf in elves.keys():
      elves[current_elf] += value
    else:
      elves[current_elf] = value

max_elf = max(elves.values())
print(max_elf)
